crates/ruff_linter/src/rules/flake8_async/rules/sleep_forever_call.rs

```diff
@@ -0,0 +1,109 @@
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{Expr, ExprCall, ExprNumberLiteral, Number};
+use ruff_python_semantic::Modules;
+use ruff_text_size::Ranged;
+
+use crate::{checkers::ast::Checker, importer::ImportRequest};
+
+/// ## What it does
+/// Checks for uses of `trio.sleep()` with >24 hour interval.
+///
+/// ## Why is this bad?
+/// `trio.sleep()` with a >24 hour interval is usually intended to sleep indefintely.
+/// This intent is be better conveyed using `trio.sleep_forever()`.
+///
+/// ## Example
+/// ```python
+/// import trio
+///
+///
+/// async def func():
+///     await trio.sleep(86401)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import trio
+///
+///
+/// async def func():
+///     await trio.sleep_forever()
+/// ```
+#[violation]
+pub struct SleepForeverCall;
+
+impl Violation for SleepForeverCall {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`trio.sleep()` with >24 hour interval should usually be `trio.sleep_forever()`.")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some(format!("Consider replacing with `trio.sleep_forever()`"))
```

Comment:
Nit: more conventional in Ruff would be "Replace with `trio.sleep_forever()`". These show up in VS Code and other editors as Quick Fix titles, so they should be short and imperative.

Reply:
Happy to update, although I'm not sure how unsafe edits appear in editors, should unsafe edits still be imperative?

Reply:
Discussed at sprints with @AlexWaygood that unsafe edits should still be imperative

```suggestion
        Some(format!("Replace with `trio.sleep_forever()`"))
```